src/utils/index.ts

```diff
@@ -46,3 +46,44 @@ export function parserUsage(parser: string, args: Arg[]) {
         console.log(`\t--${a.long},-${a.short}\t${a.description}`)
     })
 }
+
+function convertSnakeToCamel(s: string): string {
+    return s.replace(/(\_\w)/g, m => m[1].toUpperCase())
+}
+
+export function toCamel(ast: AST): AST {
+    Object
+        .keys(ast)
+        .map((n: string, i: number) => {
+            ast[n].name = convertSnakeToCamel(ast[n].name)
+            ast[n].fields
+                .map((f: GraphQLField, fi: number) => {
+                    ast[n].fields[fi].name = convertSnakeToCamel(ast[n].fields[fi].name)
+                })
```

Comment:
in this code, there's no real difference between `map` and `foreach` since you're mutating the ast. The code isn't functional in that it mutates it arguments. Consumers would be confused that:

```js
const ast = {..}
const newAst = toCamel(ast)
// now ast == newAst, oops
```

A more functional approach might look like:

```js
return Object.keys(ast).map(n => ({ 
  ...ast[n], 
  name: convertSnakeToCamel(ast[n].name), 
  fields: ast[n].fields.map(f => ({
    ...f,
    name: convertSnakeToCamel(f.name)
  })
}))
```

The object spread operator isn't available in vanilla JS, but you can use Object.assign to emulate it.

Reply:
I see your point, however function params are passed by value, not reference. so in your example `ast != newAst`

Reply:
Not sure what you mean; references are passed by value

```js
const ast = {};
const newAst = function(a) { return a }(ast);
console.log(ast == newAst);
```

prints true for me

But that's not the issue, the issue is mixing functional / imperatives styles in a way that's misleading